File: src/javakotlinexamples/java/EqualityDemo.java
package javakotlinexamples.java;

public class EqualityDemo {

    public static void main(String[] args) {
        var a = "Hallo Campus";
        var b = new String(a);

        // referential equality
        System.out.println(String.format("a == b: %b", a == b));

        // structural equality
        System.out.println(String.format("a.equals(b): %b", a.equals(b)));
    }
}

File: src/javakotlinexamples/kotlin/EqualityDemo.kt
package javakotlinexamples.kotlin

fun main(args: Array<String>) {

    val a = "Hallo Campus"
    var b = "" + a

    // structural equality
    println(String.format("a == b: %b", a == b))

    // referential equality
    println(String.format("a === b: %b", a === b))
    b = a
    println(String.format("a === b: %b", a === b))
}